gio/tests/file_monitor.rs

```diff
@@ -0,0 +1,162 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
```

Comment:
Here and elsewhere, why do you provide these tests as external integration tests instead of just as part of the implementation? The latter is faster compilation-wise

Reply:
You could add a `src/test_utils.rs` module or so that is only included with `#[cfg(test)]`